File: Atcoder_#294/ruby/c_merge_sequences.rb
def merge_sequences(n, m, a, b)
  # AとBをマージして昇順にソートし、重複を取り除いた配列
  c = (a + b).uniq.sort

  # 座標圧縮用のハッシュ
  index_map = {}
  c.each_with_index do |value, index|
    index_map[value] = index + 1
  end

  # AとBの要素をそれぞれ座標圧縮して出力
  a_indices = a.map { |ai| index_map[ai] }
  b_indices = b.map { |bi| index_map[bi] }

  puts a_indices.join(" ")
  puts b_indices.join(" ")
end

# 標準入力から値を取得
n, m = gets.split.map(&:to_i)
a = gets.split.map(&:to_i)
b = gets.split.map(&:to_i)

# 関数を呼び出し、結果を出力
merge_sequences(n, m, a, b)

File: Atcoder_#294/javascript/c_merge_sequences.js
function mergeSequences(n, m, a, b) {
    // AとBをマージして昇順にソートし、重複を取り除く
    const merged = Array.from(new Set([...a, ...b])).sort((x, y) => x - y);
    
    // 座標圧縮用のマップを作成
    const indexMap = new Map();
    merged.forEach((value, index) => {
        indexMap.set(value, index + 1);  // 1-based index
    });

    // AとBそれぞれの要素を座標圧縮して出力
    const aIndices = a.map(ai => indexMap.get(ai));
    const bIndices = b.map(bi => indexMap.get(bi));

    console.log(aIndices.join(" "));
    console.log(bIndices.join(" "));
}

// 標準入力を取得し、メイン関数を呼び出す
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8').trim().split('\n');

// 入力データをパース
const [n, m] = input[0].split(' ').map(Number);
const a = input[1].split(' ').map(Number);
const b = input[2].split(' ').map(Number);

// 関数呼び出し
mergeSequences(n, m, a, b);
